DigitalLearningSolutions.Data/Services/GroupsService.cs

```diff
@@ -0,0 +1,234 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Transactions;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Helpers;
+    using DigitalLearningSolutions.Data.Models.DelegateGroups;
+    using DigitalLearningSolutions.Data.Models.Email;
+    using DigitalLearningSolutions.Data.Models.User;
+    using MimeKit;
+
+    public interface IGroupsService
+    {
+        void SynchroniseUserChangesWithGroups(
+            DelegateUser oldDelegateDetails,

```

Comment:
Bit confused here. AccountDetailsData does not contain properties for any of the Answers, so we can't just replace the DelegateUser with it unless we want to changeAccountDetailsData to now contain those answers?
What does "Also, that's all it actually needs" mean?

Reply:
Sorry, I think I meant to put this on `LinkedFieldHelper.GetLinkedFieldChanges()`.
Ignore the comment above and I'll put something correct in the place I meant to put it.

But here you do have the same issue as on `EnrolDelegateOnGroupCourses()`, where it would be nicer for the DelegateUser to be called something like `delegateAccountWithOldDetails`.